src/base/tests/test_util.py

```diff
@@ -176,7 +179,10 @@ def test_adapter_more_domains(self):
         # double '!'
         self.mock_adapter.reset_mock()
         domain = ["!", "!", ("partner_id.user_id", "=", 1)]
-        match_domain = ["!", "!", ("partner_id.friend_id", "=", 2)]
+        if USE_ORM_DOMAIN:
+            match_domain = [("partner_id.friend_id", "=", 2)]
+        else:
+            match_domain = ["!", "!", ("partner_id.friend_id", "=", 2)]
```

Comment:
```suggestion
        match_domain = [*NOTNOT, ("partner_id.friend_id", "=", 2)]
```

Same for all other changes.